agent/utils/utils_test.go

```diff
@@ -21,86 +21,72 @@ import (
 	"github.com/aws/amazon-ecs-agent/agent/utils/ttime"
 	"github.com/aws/amazon-ecs-agent/agent/utils/ttime/mocks"
 	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
 )
 
 func TestDefaultIfBlank(t *testing.T) {
 	const defaultValue = "a boring default"
 	const specifiedValue = "new value"
 	result := DefaultIfBlank(specifiedValue, defaultValue)
-
-	if result != specifiedValue {
-		t.Error("Expected " + specifiedValue + ", got " + result)
-	}
+	assert.Equal(t, result, specifiedValue, "Expected "+specifiedValue+", got "+result)
 
 	result = DefaultIfBlank("", defaultValue)
-	if result != defaultValue {
-		t.Error("Expected " + defaultValue + ", got " + result)
-	}
+	assert.Equal(t, result, defaultValue, "Expected "+defaultValue+", got "+result)
 }
 
 func TestZeroOrNil(t *testing.T) {
-
-	if !ZeroOrNil(nil) {
-		t.Error("Nil is nil")
-	}
-
-	if !ZeroOrNil(0) {
-		t.Error("0 is 0")
-	}
-
-	if !ZeroOrNil("") {
-		t.Error("\"\" is the string zerovalue")
-	}
-
 	type ZeroTest struct {
 		testInt int
 		TestStr string
 	}
 
-	if !ZeroOrNil(ZeroTest{}) {
-		t.Error("ZeroTest zero-value should be zero")
-	}
-
-	if ZeroOrNil(ZeroTest{TestStr: "asdf"}) {
-		t.Error("ZeroTest with a field populated isn't zero")
-	}
-
-	if ZeroOrNil(1) {
-		t.Error("1 is not 0")
-	}
-
-	uintSlice := []uint16{1, 2, 3}
-	if ZeroOrNil(uintSlice) {
-		t.Error("[1,2,3] is not zero")
-	}
-
-	uintSlice = []uint16{}
-	if !ZeroOrNil(uintSlice) {
-		t.Error("[] is Zero")
-	}
-
-	if ZeroOrNil(struct{ uncomparable []uint16 }{uncomparable: []uint16{1, 2, 3}}) {
-		t.Error("Uncomparable structs are never zero")
-	}
-
-	if ZeroOrNil(struct{ uncomparable []uint16 }{uncomparable: nil}) {
-		t.Error("Uncomparable structs are never zero")
+	var strMap map[string]string
+
+	testCases := []struct {
+		param    interface{}
+		expected bool
+		name     string
+	}{
+		{nil, true, "Nil is nil"},
+		{0, true, "0 is 0"},
+		{"", true, "\"\" is the string zerovalue"},
+		{ZeroTest{}, true, "ZeroTest zero-value should be zero"},
+		{ZeroTest{TestStr: "asdf"}, false, "ZeroTest with a field populated isn't zero"},
+		{1, false, "1 is not 0"},
+		{[]uint16{1, 2, 3}, false, "[1,2,3] is not zero"},
+		{[]uint16{}, true, "[] is zero"},
+		{struct{ uncomparable []uint16 }{uncomparable: []uint16{1, 2, 3}}, false, "Uncomparable structs are never zero"},
+		{struct{ uncomparable []uint16 }{uncomparable: nil}, false, "Uncomparable structs are never zero"},
+		{strMap, true, "map[string]string is zero or nil"},
+		{make(map[string]string), true, "empty map[string]string is zero or nil"}, 
+		{map[string]string{"foo": "bar"}, false, "map[string]string{foo:bar} is not zero or nil"},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			assert.Equal(t, ZeroOrNil(tc.param), tc.expected)
```

Comment:
Doesn't hurt to be there.  In the subtest output it'll replace some characters (` ` will become `_`, for example), but the message can have the actual string.